web-components/chart/chart.tsx

```diff
@@ -0,0 +1,306 @@
+      // If we are using vegaLite, we load the data from the provided url.
+      const data = await d3csv(this.config.data.url);
+      this.dataset = data;
+      this.config.data.values = this.dataset;
+      this.config.data.url = '';
+
+      // We check to see if this chart uses interactive selections.  If yes,
+      // we figure out the list of options and populate the dropdown menu.
+      if (typeof this.config.selection == 'undefined') {
+        // If we are not using them, we compile the given config to vega
+        // and create a vega view.
+        this.compiledSpec = VegaLite.compile(this.config);
+        this.view = new Vega.View(Vega.parse(this.compiledSpec.spec)).renderer(
+          'svg'
+        );
+      } else {
+        // If we are using a selection, we get an array of the unique values in the
+        // chosen field to use to populate the dropwdown menu.
+        this.selectField = this.config.selection.select.fields[0];
+        this.selectOptions = [
+          ...new Set(this.dataset.map(item => item[this.selectField])),
+        ].sort();
+
+        // We update the config to reflect the unique values we just found
+        this.config.selection.select.bind.options = this.selectOptions;
+
+        // After updating the config, we compile it to Vega and parse it
+        this.compiledSpec = VegaLite.compile(this.config);
+
+        /* By default Vega adds and event listener to our selection that 
+          fires when the chart is clicked. The click sets the selection
+          back to null, so the user's selection gets undone. 
+
+          To get around this, we remove the event listener configuration 
+          from the compiled Vega spec.
+        */
+        const selectSignal = this.compiledSpec.spec.signals.find(
+          elem => elem.name === `select_${this.selectField}`
+        );
+        delete selectSignal.on;
+
+        // After updating the spec and compiling it, we create a new Vega view
+        // for the chart.
+        this.view = new Vega.View(Vega.parse(this.compiledSpec.spec)).renderer(
+          'svg'
+        );
+      }
+    } else {
+      // If we aren't using vegaLite, we don't need to compile the config, so we set
+      // this.compiledSpec to be the same as this.config and create the view object.
+      this.compiledSpec = this.config;
+      this.view = new Vega.View(Vega.parse(this.compiledSpec)).renderer('svg');
+    }
+  }
+
+  componentDidLoad() {
+    // Once the component loads, we initialize the view.
+    this.view.initialize(`#${this.chartID}`);
+    // Attach tooltips to the chart
+    vegaTooltip(this.view);
+
+    // If we are using selections, we build the dropdown.
+    if (this.config.selection) {
+      this.buildDropDownSelection();
+    }
+
+    // We wrap the chart svg inside a div so we can control horizontal scrolling
+    // for just the chart and not any elements around or below.
+    const chartWrapper = document.createElement('div');
```

Comment:
When there are selections, the chart component creates two children: an svg and a div to hold the select drop down in. I want just the svg to be wrapped in a div so that the overflow-x: scroll only applies to the chart and not the select element as well. Since those two elements don't exist until the chart is loading, having the wrapper element in render wasn't working for me, but would love for you to take a look.